### PR TITLE
feat: Marketplace model support in HubService

### PR DESCRIPTION
**Pull request by chrstfu**
*Description of changes:*
Adding Marketplace model support in HubService 

*Testing done:*
Testing was completed locally with example models via a BugBash. Unit tests are added to check functionality. 

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)
- [x] If adding any dependency in requirements.txt files, I have spell checked and ensured they exist in PyPi

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AWS-pratab — APPROVED** [review]

File: src/sagemaker/jumpstart/hub/utils.py
Comment (bencrabtree) [inline src/sagemaker/jumpstart/hub/utils.py:257]: Consider updating `marketplace_version` to `hub_model_version` to avoid confusion. IIUC this value is what the user passes in to `JumpStartModel`?

**Review by bencrabtree — APPROVED** [review]

File: src/sagemaker/jumpstart/hub/interfaces.py
Comment (chrstfu) [reply src/sagemaker/jumpstart/hub/interfaces.py:474]: No, not for this PR

**Review by pintaoz-aws — APPROVED** [review]

File: src/sagemaker/jumpstart/hub/utils.py
Comment (chrstfu) [reply src/sagemaker/jumpstart/hub/utils.py:256]: Ah good point, but for this utility specifically it will use the marketplace version to determine the hub model version. As a result, I'd argue that we should keep as `marketplace_version`

Comment (bencrabtree) [reply src/sagemaker/jumpstart/hub/utils.py:256]: Oh right, because we've already tried to match it with the list of hub content versions. Thanks for clarifying that makes sense!

File: src/sagemaker/jumpstart/hub/interfaces.py
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/hub/interfaces.py:454]: sanity-check: will the pySDK still understand previous schema?

File: src/sagemaker/jumpstart/hub/parser_utils.py
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/hub/parser_utils.py:22]: Inconsistent regex and method name.

Careful with renaming the method though, it would be backward incompatible.

File: src/sagemaker/jumpstart/utils.py
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/utils.py:894]: just curious, what happens if `model_id` is `None`?

File: tests/integ/sagemaker/jumpstart/utils.py
Comment (JGuinegagne) [inline tests/integ/sagemaker/jumpstart/utils.py:62]: Did you merge this to `main`? Block the release if that's the case.

Btw this code should never have been pushed to the public repo...

**Review by JGuinegagne — COMMENTED** [review]
Few post-merge clean-ups

File: src/sagemaker/jumpstart/utils.py
Comment (chrstfu) [reply src/sagemaker/jumpstart/utils.py:894]: Thanks, good catch!

File: src/sagemaker/jumpstart/hub/parser_utils.py
Comment (chrstfu) [reply src/sagemaker/jumpstart/hub/parser_utils.py:22]: True, let me revert this to also cover camelCase as well

**Pull request merged**
